models/articles.model.js

```diff
@@ -1,9 +1,85 @@
 const db = require("../db/connection");
 
-exports.selectArticleById = (articleId) => {
+exports.selectArticleById = (article_id) => {
+  const arrValues = [article_id];
   return db
-    .query("SELECT * FROM articles WHERE article_id = $1;", [articleId])
-    .then((article) => {
-      return article.rows[0];
+    .query(
+      `SELECT articles.*, Count(comments.comment_id) AS comment_count FROM articles
+    LEFT JOIN comments
+    ON comments.article_id = articles.article_id
+    WHERE articles.article_id = $1
+    GROUP BY articles.article_id;`,
+      arrValues
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: `No article found for article_id : ${article_id}`,
+        });
+      }
+      return rows[0];
     });
 };
+
+exports.updateArticleVotesById = (articleId, newVotesCount) => {
+  const arrValues = [newVotesCount, articleId];
+  return db
+    .query(
+      "UPDATE articles SET votes = votes + $1 WHERE article_id = $2 RETURNING *",
+      arrValues
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: `No article found for article_id : ${articleId}`,
+        });
+      }
+      return rows[0];
+    });
+};
+
+exports.selectArticles = (topic) => {
+  const arrValues = [topic];
+  if (topic === undefined) {
```

Comment:
You could clean up this code a bit by creating a query string using basic string concatenation, and using the `if ( topic) { ... (push values into an array, and add the WHERE clause to the query string here!)}`